src/_guides/language/concurrency/index.md

```diff
@@ -0,0 +1,403 @@
+---
+title: Concurrency in Dart
+description: [PENDING.]
+---
+
+<style>
+  img {
+    padding: 15px 0;
+  }
+}
+
+</style>
+
+Dart supports concurrent programming with async-await, isolates, and
+classes such as `Future` and `Stream`.
+Because Dart code executes in a predictable sequence that
+can’t be interrupted by other Dart code,
+sometimes Dart is called a _single-threaded language._
```

Comment:
It's wording we've used before, but I could move it to be later. (I think we might want it in the page, for SEO reasons, but it doesn't have to be at the top.)

Reply:
I don't understand what that wording means! I find it very misleading.

Why have we used that wording before? What are we trying to say with it?

Reply:
I'm a bit confused by this too. Rightly or wrongly, developers typically think of "single-threaded" as meaning "one thing at a time". 

My understanding is that isolates enable Dart code to run across multiple threads (that are hosted in different processes for isolation). 

I've seen this cause a _lot_ of confusion (e.g. https://stackoverflow.com/questions/53019756/dart-is-single-threaded-but-why-it-uses-future-objects-and-perform-asynchronous, which contains a variety of answers that are contradictory). It might be better to not introduce the phrase here, and instead have an question in the FAQ ("is Dart single-threaded?") that gives us the space to address lingering confusion head on.

Reply:
After re-reading  @munificent's [explanation](https://groups.google.com/a/dartlang.org/g/misc/c/A9eOQoLsRQM/m/8GcUQMiJAwAJ), I think it's more correct to say that in any given Dart isolate there's a single thread of execution at any given time. Dart doesn't run in parallel unless additional isolates are used.

Reply:
One could read it as saying that Dart language has no concepts of semaphores, mutexes or other tools to manage concurrent data access(that other languages like Java, C++, Python, etc provide). When writing Dart you don't have to worry about data being changed underneath you by some other Dart code that runs concurrently with you.  So that result of `bankAccount.balance += 20` will always be predictable. Perhaps having such tools available in a language makes the language "multi-threaded".

Reply:
Dart *will* utilize multiple cores even if just running a single Dart isolate. For example things like GC and async IO will utilize other cores. I got this confirmed by Slava today, and i think its aligned with what Bob says in that link also.

Dart really isn't "single treaded".

Reply:
To @johnpryan's comment: "Dart doesn't run in parallel unless additional isolates are used", couldn't you make an equal argument that "C code doesn't run in parallel unless additional forks are spawned"? That is, the language is only single threaded if you don't use threading features :) 

Maybe the confusion is that some of us are thinking of the phrase "single-threaded" as a positive (you don't have to worry about things changing under you, code executes in a predictable sequence). The reason I'm sensitive to the phrase is that a lot of folk mistakenly think of it as indicating a deficit in Dart today...

Reply:
> Why have we used that wording before? What are we trying to say with it?

also page 9 of Gilad's and Lars's [presentation at Goto conference](https://gotocon.com/dl/goto-aarhus-2011/slides/GiladBracha_and_LarsBak_OpeningKeynoteDartANewProgrammingLanguageForStructuredWebProgramming.pdf) might give some context ```SO WHAT IS DART?
...
• Single-threaded
...
```

Reply:
Maybe we could say each isolate has a single call stack, heap, and event loop?